File: app/src/main/java/com/rafaelfiume/prosciutto/adviser/ProductDescriptionParser.java
package com.rafaelfiume.prosciutto.adviser;

import android.util.Log;

import com.rafaelfiume.prosciutto.adviser.domain.ProductDescription;

import static com.rafaelfiume.prosciutto.adviser.util.Xml.getValueFrom;
import static com.rafaelfiume.prosciutto.adviser.util.Xml.xmlFrom;

public class ProductDescriptionParser {

    public ProductDescription parse(String xml) {
        try {
            return new ProductDescription(getValueFrom(xmlFrom(xml), "//extract"));
        } catch (Exception e) {
            Log.e(ProductDescriptionParser.class.getName(), "error when parsing xml " + xml, e);
            throw new RuntimeException("could not parse advice xml: " + xml, e);
        }
    }

}

File: app/src/main/kotlin/com/rafaelfiume/prosciutto/adviser/ProductDescriptionParser.kt
package com.rafaelfiume.prosciutto.adviser

import android.util.Log

import com.rafaelfiume.prosciutto.adviser.domain.ProductDescription

import com.rafaelfiume.prosciutto.adviser.xml.Xml.getValueFrom
import com.rafaelfiume.prosciutto.adviser.xml.Xml.xmlFrom

class ProductDescriptionParser {

    fun parse(xml: String): ProductDescription {
        try {
            return ProductDescription(getValueFrom(xmlFrom(xml), "//extract"))
        } catch (e: Exception) {
            Log.e(ProductDescriptionParser::class.java.name, "error when parsing xml $xml", e)
            throw RuntimeException("could not parse advice xml: $xml", e)
        }

    }

}
